test/common/filter/tcp_proxy_test.cc

```diff
@@ -47,13 +74,235 @@ TEST(TcpProxyConfigTest, BadTcpProxyConfig) {
       Json::Exception);
 }
 
+TEST(TcpProxyConfigTest, Routes) {
```

Comment:
Do we have a test for the no route case? We should verify that connection gets closed, and stat gets incremented.